test/sequential/market.nonTransferable.multiswap.test.ts

```diff
@@ -0,0 +1,294 @@
+    poolIds.push(poolId);
+    i++;
+  }
+  return poolIds;
+}
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+
+  await setupApi();
+  await setupUsers();
+  sudo = getSudoUser();
+  testUser = "0x798d4ba9baf0064ec19eb4f0a1a45785ae9d6dfc";
+  api = getApi();
+});
+
+beforeEach(async () => {
+  let foundationMembers: any;
+
+  foundationMembers = await FoundationMembers.getFoundationMembers();
+
+  [token1, token2, token3, token4, token5, token6] =
+    await Assets.setupUserWithCurrencies(
+      sudo,
+      [BN_BILLION, BN_BILLION, BN_BILLION, BN_BILLION, BN_BILLION, BN_BILLION],
+      sudo,
+    );
+
+  const oldFounder2 = foundationMembers[2];
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      oldFounder2,
+      api.tx.foundationMembers.changeKey(testUser),
+    ),
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  foundationMembers = await FoundationMembers.getFoundationMembers();
+  expect(foundationMembers).toContain(testUser);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudo(api.tx.tokens.mint(token1, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token2, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token3, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token4, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token5, testUser, BN_HUNDRED_BILLIONS)),
+    Sudo.sudo(api.tx.tokens.mint(token6, testUser, BN_HUNDRED_BILLIONS)),
+  );
+  const poolEvents = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(token1, BN_MILLION, token2, BN_MILLION, "StableSwap"),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(
+        token2,
+        BN_MILLION,
+        GASP_ASSET_ID,
+        BN_MILLION,
+        "StableSwap",
+      ),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(
+        GASP_ASSET_ID,
+        BN_MILLION,
+        token3,
+        BN_MILLION,
+        "StableSwap",
+      ),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(token3, BN_MILLION, token4, BN_MILLION, "StableSwap"),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(token4, BN_MILLION, token5, BN_MILLION, "StableSwap"),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      Market.createPool(token5, BN_MILLION, token6, BN_MILLION),
+    ),
+  );
+  const eventResponse = getEventResultFromMangataTx(poolEvents);
+  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  liqIds = await getStablePoolId(poolEvents);
+  await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser,
+      api.tx.foundationMembers.changeKey(oldFounder2),
```

Comment:
reuse the foundarionMembers class from previous PR :)

Reply:
Done